app/models/assignment.rb

```diff
@@ -826,7 +826,10 @@ def is_criteria_mark?(ta_id)
     assign_graders_to_criteria && self.criterion_ta_associations.where(ta_id: ta_id).any?
   end
 
-  def get_num_assigned(ta_id = nil)
+  def get_num_assigned(ta_id = nil, bulk: false)
+    if bulk
+      return @cache_ta_results[ta_id][:total_results].size
```

Comment:
In general, no methods except `cache_ta_results` should access the instance attribute `@cache_ta_results`, since it might not be defined.

But, what you should do is ensure `cache_ta_results` always *returns* the correct hash (i.e., `@cache_ta_results`); that way it can be used anywhere you want to use the instance attribute. Basically, the instance attribute is private but the method `cache_ta_results` is public.

So in this case, you can write `return cache_ta_results[ta_id][:total_results].size` instead.

Reply:
But also, direct access is bad: use `Hash#dig` instead, as you do elsewhere.

Reply:
Finally, for the `bulk: true` option, only do it when `ta_id` is not `nil`. That is, move this into the `else` branch below.